lib/land/trackers/api_tracker.rb

```diff
@@ -202,11 +202,13 @@ def user_agent
                      Land.config.blank_user_agent_string
 
         @user_agent = UserAgent[user_agent]
+        @user_agent.update(user_agent_type: UserAgentType['user'])
 
         if Land.config.identify_crawlers && defined?(CrawlerDetect)
           crawler_detect = CrawlerDetect.new(user_agent)
-          user_agent_type = crawler_detect.is_crawler? ? 'crawl' : 'api'
-          @user_agent.update(user_agent_type: UserAgentType[user_agent_type])
+          if crawler_detect.is_crawler?
+            @user_agent.update(user_agent_type: UserAgentType['crawl'])
+          end
```

Comment:
We are effectively making 3 database calls in this method now, and all the update calls will not throw an error or be noisy. 

Changes needed: 
- change all `#update` calls to use setters `@user_agent.user_agent_type = UserAgentType['user']`
- use a `#save!` method to save, rescue any errors and log to datadog.
- refactor the logging of the errors to a method (refactor common error logging to a method)